core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/hashJoin/HashJoinOpExec.scala

```diff
@@ -82,7 +83,7 @@ class HashJoinOpExec[K](
 
             storedTuples.foreach(buildTuple => {
               val builder = Tuple
-                .newBuilder()
+                .newBuilder(outputSchema)
```

Comment:
Are you sure this is changing anything? I am adding a buildTuple and I think the schema of the build tuple will be used.

Reply:
Oh I see your above change now. If schema is already there in tupleBuilder, it will ignore any new schema info. This again is confusing. I am inputting tuple with some schema but that schema will be ignored because schema is already defined in TupleBuilder

Reply:
I could add a check to see whether the outSchema and the tuple's schema are the same. Or change `newBuilder` to maybe `newBuilderWithSchema`. But it's a change that's required to reduce the number of schema objects.